src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/Internal/ConditionsCallout/ConditionsCallout.state.ts

```diff
@@ -0,0 +1,68 @@
+import produce, { current } from 'immer';
+import {
+    deepCopy,
+    getDebugLogger
+} from '../../../../../../Models/Services/Utils';
+import { isStateDirty } from './ConditionCalloutUtility';
+import {
+    ConditionCalloutAction,
+    ConditionCalloutActionType,
+    ConditionCalloutReducerType,
+    IConditionCalloutState
+} from './ConditionsCallout.types';
+
+const debugLogging = true;
+const logDebugConsole = getDebugLogger('Condition', debugLogging);
+
+export const defaultConditionCalloutState: IConditionCalloutState = {
+    originalCondition: null,
+    conditionToEdit: null,
+    isDirty: false
+};
+
+export const ConditionCalloutReducer: ConditionCalloutReducerType = produce(
+    (draft: IConditionCalloutState, action: ConditionCalloutAction) => {
+        logDebugConsole(
+            'info',
+            `Updating ConditionCallout ${action.type} with: {payload, visualRule}`,
+            (action as any).payload, // ignore that payload doesn't always come since this is just a log
+            current(draft.conditionToEdit)
+        );
+        switch (action.type) {
+            /** Condition actions */
+            case ConditionCalloutActionType.FORM_CONDITION_COLOR_SET: {
+                draft.conditionToEdit.visual.color = action.payload.color;
+                break;
+            }
+            case ConditionCalloutActionType.FORM_CONDITION_ICON_SET: {
+                draft.conditionToEdit.visual.iconName = action.payload.iconName;
+                break;
+            }
+            case ConditionCalloutActionType.FORM_CONDITION_LABEL_SET: {
+                draft.conditionToEdit.visual.labelExpression =
+                    action.payload.label;
+                break;
+            }
+            case ConditionCalloutActionType.FORM_CONDITION_VALUES_SET: {
+                draft.conditionToEdit.values = action.payload.values;
```

Comment:
Values can either be `[true/false]`, `[number, number]` or `[strings...]`. I think for the string case it could make sense but don't see harm in keeping it as a single dispatch and using the set pattern.